Tests/BackendIntegrationTests/StoreKitIntegrationTests.swift

```diff
@@ -314,6 +314,16 @@ class StoreKit1IntegrationTests: BaseStoreKitIntegrationTests {
         expect(eligibility) == .eligible
     }
 
+    func testIneligibleForIntroForDifferentProductInSameSubscriptionGroupAfterPurchase() async throws {
+        let product1 = try await self.annualPackage.storeProduct
+        let product2 = try await self.annualNoIntroProduct
+
+        _ = try await Purchases.shared.purchase(product: product2)
+
+        let eligibility = await Purchases.shared.checkTrialOrIntroDiscountEligibility(product: product1)
+        expect(eligibility) == .ineligible
```

Comment:
> A customer reported that they are calling `checkTrialOrIntroDiscountEligibility` and are receiving eligible for a trial product within the same subscription group of their current subscription. The payment sheet from Apple confirms that they should be receiving ineligible

Both SK1 and SK2 agree here that this is `eligible`, confirming that this could be an Apple bug.

Reply:
Maybe add it as a comment so everyone knows why this is failing

Reply:
I left some comments on the jira ticket:

This should be ineligible IFF the purchase is active. Do we still have the bug where min time rate means instantly-expiring subscriptions? If so, then that might explain it. 

Also, we need to update the code for SK1 to account for that, it currently doesn't take into account a possibly active subscription. 

Maybe we can add that _other_ use case too? i.e.: purchase with no intro, cancel, expire, check that is eligible for different product in same subs group that has intro trial

Reply:
so... that's all to say that this test case is valid, but maybe we could add a line that checks that the subscription is currently active to make it more explicit? or rename the test case to include the fact that the subs is active?

Reply:
Yeah I'll add another test.
This one is failing so we can't keep it.

I'll fix the SK1 logic, and temporarily `skip` the test for SK2.

Reply:
> This should be ineligible IFF the purchase is active. Do we still have the bug where min time rate means instantly-expiring subscriptions? If so, then that might explain it.

No we don't case we moved from weekly to monthly subscriptions (and we add a debug (warning?) log to mention that this is happening).

Reply:
> Maybe we can add that other use case too? i.e.: purchase with no intro, cancel, expire, check that is eligible for different product in same subs group that has intro trial

Done:

```swift
func testEligibleForIntroForDifferentProductInSameSubscriptionGroupAfterExpiration() async throws {
    let product1 = try await self.annualNoIntroProduct
    let product2 = try await self.annualPackage.storeProduct

    let customerInfo = try await Purchases.shared.purchase(product: product1).customerInfo
    let entitlement = self.verifyEntitlementWentThrough(customerInfo)

    try await self.expireSubscription(entitlement)

    let eligibility = await Purchases.shared.checkTrialOrIntroDiscountEligibility(product: product2)
    expect(eligibility) == .eligible
}
```

Reply:
That test passes on both SK1 and SK2.